pkg/repo/chartrepo.go

```diff
@@ -133,10 +133,21 @@ func (r *ChartRepository) DownloadIndexFile() (string, error) {
 		return "", err
 	}
 
-	if _, err := loadIndex(index); err != nil {
+	indexFile, err := loadIndex(index)
+	if err != nil {
 		return "", err
 	}
 
+	// Create the chart list file in the cache directory
+	var charts strings.Builder
+	for name := range indexFile.Entries {
+		fmt.Fprintln(&charts, name)
```

Comment:
would you mind writing a unit test for this if possible?

Reply:
Will do. Thanks for the review!

Reply:
The simplest way to test this was to extend the Index file test of `index_test.go`.
The latest version of the PR does just that.